### PR TITLE
Hook Objects

### PR DESCRIPTION
**Pull request by ohsnapitscolin**
### Description
Returns objects contain `loading`, `error`, `doc(s)` and `data` from the `useQuery`, `useDoc` and `useDocs` hooks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,6 +99,13 @@ const doc = collection.doc('my-id');
...
+The hook returns an object that includes the following properties:
+
+- `loading`: Whether data has been returned by the query.
+- `docs`: The query results as an array of document references.
+- `data`: The query results as an array of data (same as calling `docs.map(d => d.data)`).
+- `error`: The error object, if an error occurs while executing the query.
+
 ```jsx
 function App() {
   const collection = useCollection('my-collection');
```
Comment (peledni) [inline README.md:111]: maybe make this a typescript example? (add types)

File: README.md [hunk 2]
```diff
@@ -128,42 +135,55 @@ streamed.
...
+particular document.
+
+The hook returns an object that includes the following properties:
+
+- `loading`: Whether data has been returned by the document query.
+- `doc`: The document reference.
+- `data`: The document data (same as calling `doc.data`). This can be undefined if no data has been received or if the document has been deleted. 
+- `error`: The error object, if an error occurs while querying for the document.
 
 ```jsx
```
Comment (peledni) [inline README.md:148]: here too (typescript)

File: README.md [hunk 3]
```diff
@@ -211,6 +231,16 @@ changes, the current observable will unsubscribe, and a new subscription will be
 can see the `where` condition of the query relies on the `bar` variable. To ensure that the query is properly updated
 when `bar` changes, we need to pass it as a dependency.
 
+Whenever the `deps` change, `loading` is reset to true until a value is emitted from the newly created observable.
+
+In addition to passing an observable directly to the `useObservable` hook, you can also pass a function that return an observable.
```
Comment (peledni) [inline README.md:236]:
```suggestion
In addition to passing an observable directly to the `useObservable` hook, you can also pass a function that returns an observable.
```

**Review by peledni — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (ohsnapitscolin) [reply README.md:111]: Done. Once the collection is typed the types carry through everywhere else.

**Pull request merged**